### PR TITLE
Correction on prev_token, the index should decremented

### PR DESCRIPTION
**Pull request by ivanceras**
This change is [Reviewable](https://reviewable.io/reviews/andygrove/sqlparser-rs/20)

### CODE REVIEW

File: src/dialect.rs [hunk 1]
```diff
@@ -383,6 +383,91 @@ impl Dialect for GenericSqlDialect {
...
+    fn is_identifier_part(&self, ch: char) -> bool {
+        (ch >= 'a' && ch <= 'z')
+            || (ch >= 'A' && ch <= 'Z')
+            || (ch >= '0' && ch <= '9')
+            || ch == '@'
+            || ch == '_'
+    }
+}
+
+pub struct PostgreSqlDialect {}
```
Comment (andygrove) [inline src/dialect.rs]: Could you move the postgres dialect to a new source file please

**Comment by ivanceras** [comment]
I have moved the dialects into their own submodules.

**Comment by ivanceras** [comment]
The code coverage interpretation seems to be wrong. Take a loot at the sqlparser.rs, the test is down to 75%, where in fact, I just merely move the test files into sqlparser/test.rs but still under the sqlparser module. The sqlast module, doesn't need to be tested as the code there are faily trivial.

**Comment by ivanceras** [comment]
Most of my projects involve postgresql, and some sqlite.
I was able to use this sqlparser to parse pg_dump files already.
I haven't thought much of organizing the dialect specific code yet. I'll take a look at what you could come up and maybe evolve from there. 

I did started working on each separate branch for each PR, but I unknowingly merge them back to master without waiting for the PR to get merged. I forgot that github PR is per branch and follow up commits will become parts of the previous PR. I will try to make a separate branch for each PR and resist the urge to always work in master, as with my other repositories.

> I hope you take all of this as constructive criticism! I appreciate the contribution and hope you will help me make this the leading SQL parser library for Rust!

Thank you for your consideration. I've been looking for a good sql parser but the use of parser-generator makes it hard for me to understand how works. I take a look at this code with a written parser and it immediately make sense to me.

**Pull request merged**